### ISSUE TITLE
LightGBMDelegate is not available for PySpark users

### ISSUE THREAD

**Issue by denmoroz** [microsoft/SynapseML#1499]
**Is your feature request related to a problem? Please describe.**
`LightGBMDelegate` is not available for PySpark users

**Describe the solution you'd like**
`from synapse.ml.lightgbm import LightGBMDelegate`

**Additional context**
As described in https://github.com/microsoft/SynapseML/issues/1094#issuecomment-1103781493

I am using hyperopt to tune booster parameters with `average_precision` as a metric using CV. From the code I see that in Scala I can override [afterTrainIteration](https://github.com/microsoft/SynapseML/blob/339e516e1c38d583c95c95d15d6b09fd86e6a43b/lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/LightGBMDelegate.scala#L48) and gather metrics from `validEvalResults` to pass it then to hyperopt fmin / fmax.

But at the same delegate is not available for pyspark users, so what I really have to do is:

1. use LightGBM internal `average_precision` for early stopping
2. then gather predictions for validation set(s) and calculate average precision once again by my own in hyperopt objective

It works, but for sure it is not optimal and it would be so much easier (and safer) just to use LightGBM's output.

AB#1776895

**Comment by niehaus59** [comment]
Hi denmoroz - We have been discussing this and while the idea is good, we will need to have a bigger discussion about a more general hook facility that would cover the majority of SynapseML, if not all of it, and make that available in python and R.  Making LightGBMDelegate available in python would complicate that effort.  Please respond here if you have concerns.

**Comment by denmoroz** [comment]
@niehaus59 Thanks for your response!

Am I getting this right: **there is** a plan (or intention at least) to make a more general solution and expose most of SynapseML to Python / R users?

**Comment by denmoroz** [comment]
Also I see that there is ROC AUC in `ComputeModelStatistics`, but at the same time Average Precision (areaUnderPR) is absent. It will be awesome to add it also as it is very useful for many binary classification tasks.

**Comment by denmoroz** [comment]
@niehaus59
Thanks for explanation!

> For the ROC AUC question, it would be best to start a new issue to cover that

Sure
